table.go

```diff
@@ -509,14 +512,17 @@ func (t *Table) SchemaIterator(
 	tx uint64,
 	pool memory.Allocator,
 	iterOpts logicalplan.IterOptions,
-	iterator func(ctx context.Context, r arrow.Record) error,
+	iterators []func(ctx context.Context, r arrow.Record) error,
 ) error {
 	ctx, span := t.tracer.Start(ctx, "Table/SchemaIterator")
 	span.SetAttributes(attribute.Int("physicalProjections", len(iterOpts.PhysicalProjection)))
 	span.SetAttributes(attribute.Int("projections", len(iterOpts.Projection)))
 	span.SetAttributes(attribute.Int("distinct", len(iterOpts.DistinctColumns)))
 	defer span.End()
 
+	// TODO(metalmatze): Wrap the following in an errgroup to introduce concurrency
+	iterator := iterators[0]
+
 	rowGroups, err := t.collectRowGroups(ctx, tx, iterOpts.Filter)
```

Comment:
If we're adding concurrency it probably makes sense to have the row group collection be writers into a channel instead of in-memory buffering all of them.

Reply:
Yes, that would be perfect. We could then probably use channels instead of the errgroup altogether, right?